worker-batch-ingestion-plugin/src/main/java/com/github/cafdataprocessing/worker/ingestion/DocumentWorkerDocumentDeserializer.java

```diff
@@ -78,57 +77,57 @@ public DocumentWorkerDocument deserialize(
     private DocumentWorkerDocument deserializeDocumentWorkerDocument(
         final JsonParser jsonParser,
         final DeserializationContext deserializationContext,
-        final JsonToken currentJsonToken,
         final MutableInt totalSubdocuments
     ) throws IOException
     {
-        if (currentJsonToken != JsonToken.START_OBJECT) {
+        if (jsonParser.currentToken() != JsonToken.START_OBJECT) {
             throw new IllegalStateException(
                 String.format("Expected '{' at %s", jsonParser.getCurrentLocation().toString()));
         }
-
-        jsonParser.nextToken();
-
+        
         final DocumentWorkerDocument documentWorkerDocument = new DocumentWorkerDocument();
 
+        jsonParser.nextToken();
+        
         while (jsonParser.currentToken() != JsonToken.END_OBJECT) {
-            if (jsonParser.currentToken() == JsonToken.FIELD_NAME) {
-                final String field = jsonParser.getCurrentName();
-                jsonParser.nextToken();
-                switch (field) {
-                    case "reference": {
-                        documentWorkerDocument.reference = jsonParser.getValueAsString();
-                        break;
-                    }
-                    case "fields": {
-                        documentWorkerDocument.fields = deserializationContext.readValue(
+            if (jsonParser.currentToken() != JsonToken.FIELD_NAME) {
```

Comment:
Is this check necessary? A Json parsing exception might be thrown earlier on @ line 90 :jsonParser.nextToken();

Reply:
This is protecting against bugs, prior versions of this change were incorrectly skipping elements and this check is protecting against that.